debezium-core/src/main/java/io/debezium/transforms/time/TimestampConverter.java

```diff
@@ -0,0 +1,785 @@
+
+        OPTIONAL_TRANSLATORS.put(TYPE_STRUCT, SchemaBuilder.type(Schema.Type.STRUCT).optional().build().schema());
+
+    }
+
+    /**
+     * Static method to check if a particular field is of STRUCT type or not.
+     *
+     * @return true for struct field and false for others
+     */
+    protected static Boolean isStruct(Field field) {
+        return field.schema().type().getName().equals("struct") ? true : false;
+    }
+
+
+    /**
+     * Static method to generate the database.table name from input string.
+     *
+     * @param schemaName
+     * @return database.tablename.
+     */
+
+    protected static String databaseTableString(String schemaName) {
+        return schemaName.substring(schemaName.indexOf(".") + 1).replace(schemaName.substring(schemaName.lastIndexOf(".")), "");
+    }
+
+    /**
+     * * Static method to provide a map containing Key(field) and list of Values(nested fields associated to the field and if field is not a struct/nor has nested fields like ts_ms, then entry in map is created a ts_ms -> [None]).
+     *
+     * @param tableName
+     * @return Map containing key as field and value as list of nested field names.
+     */
+    protected static Map<String, List<String>> getFields(String tableName) {
+        return FieldConversions.get(tableName) != null ? FieldConversions.get(tableName).keySet().stream().collect(Collectors.
+                groupingBy(key -> key.split("\\.")[0], Collectors.mapping(val -> val.split("\\.").length == 2 ? val.split("\\.")[1].trim() : "None", Collectors.toList())
+                )
+        ) : Collections.emptyMap();
+    }
+
+    /**
+     * Static method to sanitize the name of the table from STRUCT schema, example schema.name=SERVERNAME.DATABASENAME.TABLENAME.OTHERS is provided and the method return just the DATABASENAME.TABLENAME.
+     *
+     * @return DATABASENAME.TABLENAME string.
+     */
+    protected static Set<String> getTableNames() {
+        return FieldConversions.keySet();
+    }
+
+    /**
+     * Static method to return the fields that are not nested, for example ts_ms , would search based on the None value that we insert using the getfields methods.
+     */
+    protected static List<String> findFlatFields(String tableName) {
+        List<String> flatFieldsArray = new ArrayList<>();
+        for (Map.Entry<String, List<String>> flatFields : getFields(tableName).entrySet()) {
+            if (flatFields.getValue().size() == 1 && flatFields.getValue().stream().anyMatch(x -> x.equals("None"))) {
+                flatFieldsArray.add(flatFields.getKey());
+            }
+        }
+        return flatFieldsArray;
+    }
+
+    // This is a bit unusual, but allows the transformation config to be passed to static anonymous classes to customize
+    // their behavior
+    public static class Config {
+
+        public final String[] field;
+        public final SimpleDateFormat format;
```

Comment:
`SimpleDateFormat` isn't thread-safe. I'm not sure whether this config may be shared between threads (will `Transform` instances be called concurrently from multiple threads potentially?), but it may be safer to just share the String format and create new SDF instances when required.